### ISSUE TITLE
@typescript-eslint/parser don't work properly with symlink folders

### ISSUE THREAD

**Issue by btd1337** [typescript-eslint/typescript-eslint#2234]
**Repro**

I use a symlink from my project folder point to my home folder:

`/opt/project/workspace` => `~/worspace`

Opening the VSCode using the symlink path `~/workspace` the error always persists.

Opening VSCode using the original path: `/opt/project/workspace` solves the problem.

This shouldn't be an error, but for now it is.

It's important to say that I made all other issues solutions related but nothing works.

```JSON
{
  "rules": {
   '@typescript-eslint/ban-ts-comment': 'warn',
   '@typescript-eslint/no-unused-vars': ['error', { args: 'none' }]
  },
  "parserOptions": {
    project: 'tsconfig.eslint.json',
    sourceType: 'module',
    ecmaVersion: 6,
  }
}
```

```ts
// tsconfig.eslint.json
{
	"extends": "./tsconfig.json",
	"include": [
		"./src/**/*.ts",
		"./test/**/*.ts",
	]
}

```

```TS
// your repro code case
```

**Expected Result**
Code below works properly:

```ts
import { Response } from 'express';
```

**Actual Result**

```log
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: "ANY FILE WITH IMPORT".
The file must be included in at least one of the projects provided.
```

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `^3.2.0` |
| `TypeScript`                | `3.9.5` |
| `ESLint`                    | `7.2.0` |
| `node`                      | `12.13.1` |
| `npm`                       | `6.14.4` |

**Comment by bradzacher** [comment]
Does the lint work fine from the CLI when it is in the symlinked folder? or does it only error from vscode when it's opened in the symlinked folder?

Could you please set `parserOptions.debugLevel: true` and dump the output?

**Comment by btd1337** [comment]
Enabling `parserOptions.debugLevel: true` on my `.eslintrc.js` and open symlink folder:

![Captura de tela de 2020-06-20 22-35-40](https://user-images.githubusercontent.com/11943860/85214810-76c62080-b346-11ea-9a19-f2638485891e.png)

Open from original folder location:

![Captura de tela de 2020-06-20 22-38-43](https://user-images.githubusercontent.com/11943860/85214842-d290a980-b346-11ea-8129-0056c3587c31.png)

Just point error in .eslintrc because it not be included in tsconfig file.

Test it on your machine and watch the error happen.

The lint always looks for the relative path from the beginning of the original path and gives an error when it is opened by a symbolic path.

Google translate sorry.

**Comment by bradzacher** [comment]
Does the lint work fine from the CLI when it is in the symlinked folder? or does it only error from vscode when it's opened in the symlinked folder?

> Test it on your machine and watch the error happen.

I don't have your setup, and I don't have the bandwidth to recreate it myself right now.
So you need to help me investigate within your setup.

With the debug flag turned on, you need to get the logs out of the extension, which you can do via the "eslint: show output channel" in vscode.

**Comment by btd1337** [comment]
Yeah, lint works fine from the CLI. The bug occurs only into VSCode.

Running the command `eslint: show output channel`: 

```
[Info  - 22:19:24] ESLint server stopped.
[Info  - 22:19:25] ESLint server running in node v12.8.1
[Info  - 22:19:25] ESLint server is running.
[Info  - 22:19:26] ESLint library loaded from: /opt/dev/back/node_modules/eslint/lib/api.js   // full path
```

I'm open in `~/project/` and eslint point to ` /opt/dev/back/`
